pkg/cli/debug/debug.go

```diff
@@ -848,6 +848,11 @@ func (o *DebugOptions) approximatePodTemplateForObject(object runtime.Object) (*
 				},
 			},
 		}, nil
+	}
+	if len(o.ToNamespace) > 0 {
```

Comment:
Nope, we need this for everything else too.

Reply:
ok, my first go was to clear the Volumes/VMounts, but then still saw fails w/ ServiceAccounts, so I'll clear both when ToNamespace.

Reply:
I've updated